krun/platform.py

```diff
@@ -459,9 +461,7 @@ def throttle_lines(dct):
 
             log_and_mail(self.mailer, warn, "throttle counts changed", body,
                          manifest=manifest)
-            return True
-        else:
-            return False
+            raise util.RerunExecution("throttle counters changed")
```

Comment:
Out of interest, why do this as an exception? It feels like a bit of a sledgehammer!

Reply:
Ah, I think I see the answer: because we already have `RerunException` elsewhere?

Reply:
Yeah, maybe, but that's kind of baked in at this point.

I think it's because it allows us to propagate an error (and a message) several frames up without having to explicitly pass it around.

I don't think we should change this now.

Reply:
I agree.

Reply:
> because we already have RerunException elsewhere?

That's right, I'm hooking into existing re-run logic.